File: src/athena/core/thread_models.py
"""Models describing Athena agent tasks, threads, and turns."""

from pydantic import BaseModel

from athena.core.contracts import ArtifactRef


class AthenaThread(BaseModel):
    """对话线程 — 对应一个 Agent 会话，通过 artifact 引用持久化上下文。"""

    thread_id: str
    session_id: str
    status: str
    context_ref: ArtifactRef


class AthenaTurn(BaseModel):
    """单轮对话 — 包含请求引用和可选的执行结果引用。"""

    turn_id: str
    thread_id: str
    request_ref: ArtifactRef
    status: str
    result_ref: ArtifactRef | None = None

File: athena_ts/packages/athena-core/src/models/thread-models.ts
import { z } from "zod"
import { ArtifactRef } from "./contracts.js"

/** 对话线程——对应一个 Agent 会话。 */
export const AthenaThreadSchema = z.object({
  thread_id: z.string(),
  session_id: z.string(),
  status: z.string(),
  context_ref: ArtifactRef,
})
export type AthenaThread = z.infer<typeof AthenaThreadSchema>

/** 单轮对话——请求引用与可选执行结果引用。 */
export const AthenaTurnSchema = z.object({
  turn_id: z.string(),
  thread_id: z.string(),
  request_ref: ArtifactRef,
  status: z.string(),
  result_ref: ArtifactRef.nullable().default(null),
})
export type AthenaTurn = z.infer<typeof AthenaTurnSchema>
